sys/phydat/phydat_str.c

```diff
@@ -78,14 +80,35 @@ const char *phydat_unit_to_str(uint8_t unit)
     }
 }
 
-char phydat_scale_to_str(int8_t scale)
+char phydat_scale_to_str(int8_t scale, uint8_t unit)
 {
+    /* extra handling for multi and non dimensional units */
+    switch (unit) {
+        case UNIT_M2:       /* rescale for 2d unit */
+            if (scale % 2 != 0) {
```

Comment:
a general problem with `scale_to_str` is that it cannot handle a scaling factor that`s not a multiple of 3. But it would be nice to convert something like `{.val = {2, 0, 0}, .unit = UNIT_M, .scale = 5}` into `200km` currently it outputs `20000m`